merge 2020.06.x

yagee-de opened the pull request.

dependabot[bot] committed:
Bump ini from 1.3.5 to 1.3.7 in /mycore-mets

Bumps [ini](https://github.com/isaacs/ini) from 1.3.5 to 1.3.7.
- [Release notes](https://github.com/isaacs/ini/releases)
- [Commits](https://github.com/isaacs/ini/compare/v1.3.5...v1.3.7)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump ini from 1.3.5 to 1.3.7 in /mycore-webtools

Bumps [ini](https://github.com/isaacs/ini) from 1.3.5 to 1.3.7.
- [Release notes](https://github.com/isaacs/ini/releases)
- [Commits](https://github.com/isaacs/ini/compare/v1.3.5...v1.3.7)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump ini from 1.3.5 to 1.3.8 in /mycore-classeditor

Bumps [ini](https://github.com/isaacs/ini) from 1.3.5 to 1.3.8.
- [Release notes](https://github.com/isaacs/ini/releases)
- [Commits](https://github.com/isaacs/ini/compare/v1.3.5...v1.3.8)

Signed-off-by: dependabot[bot] <support@github.com>

toKrause committed:
MCR-2331 Include email in user search

Adds an additional parameter to MCRUserManager#listUsers and MCRUserManager#countUsers that allows to specify a search pattern to be used for searchs by email.
Uses this parameter in MCRUserServlet#listUsers using the provided search terms as a search pattern.

sebhofmann committed:
MCR-2332 commands to load update classification from uri

sebhofmann committed:
MCR-2334 MCR.Viewer.canvas.overview.enabled has no effect

sebhofmann committed:
MCR-2329 systemid entry for xml-2009.xsd in mycore-pi is missing

OPNA2608 committed:
MCR-2315 Common methods for {base,type,project} CLI command variants (#1117)

Includes trivial migrations of various commands to these new helper methods.

OPNA2608 committed:
MCR-2315 Common methods for {base,type,project} CLI command variants

Fix inverted Base-ID check in MCRCommandUtils

kkrebs committed:
MCR-2140 return element always with position bracket to make it unique (#1120)

* MCR-2140 Use position predicate only if element is not unique
* MCR-2140 Removed useless namespace declaration
* MCR-2140 Be more verbose when test fails
* MCR-2140 Updated expected output in tests
* MCR-2140 Fixed use of absolute path in MCRBinding
* MCR-2140 Fix checking resubmitted repeated values

Co-authored-by: Frank Lützenkirchen <frank.luetzenkirchen@uni-due.de>

yagee-de committed:
MCR-2195 leave directory handling to commons-compress

yagee-de committed:
MCR-2156 fix last modified time is missing in tar file

If directory is missing in TAR file we cannot restore
the original last modified time. Log a warning instead
of throwing an exception later on.

yagee-de committed:
MCR-2301 fix handling of file parameter

check if parameter is sane and ready to be used

rsteph-de committed:
MCR-2293 replace colon in XSLT parameter names from properties

yagee-de committed:
MCR-2087 add XSL files from config directory

yagee-de committed:
MCR-2186 do not depend on IPv4 to generate id prefix

yagee-de committed:
MCR-2328 fix 'parameters' value

yagee-de committed:
MCR-2321 encode image path in Location header

yagee-de committed:
MCR-2100 deprecate MCRHIBConnection.getSession()

sebhofmann committed:
MCR-2257 browsing arrows in viewer doesn't work in all cases

sebhofmann committed:
MCR-2342 revert to dojo 1.6.5

yagee-de committed:
MCR-2341 fixed xsl:iterate in translate-with-params

rsteph-de committed:
MCR-2291 urlencode OAI resumption token

sebhofmann committed:
MCR-2329 missing systemid entry for xml-2009.xsd for mycore-pi is missing

MyCoRe-Travis commented:
Something gone wrong on 553fe504ae32e6858ab976cf3c19adc170769141! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6472-pull_request-1159

MyCoRe-Travis commented:
Something gone wrong on eb8e25497d2d14fb7a7518b9c565a24081dad03b! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6476-pull_request-1159

dependabot[bot] committed:
Bump ini from 1.3.5 to 1.3.8 in /mycore-wcms2

Bumps [ini](https://github.com/isaacs/ini) from 1.3.5 to 1.3.8.
- [Release notes](https://github.com/isaacs/ini/releases)
- [Commits](https://github.com/isaacs/ini/compare/v1.3.5...v1.3.8)

Signed-off-by: dependabot[bot] <support@github.com>

MyCoRe-Travis commented:
Something gone wrong on 33e80f60d921bc60642a01fb885aedadad1f7b92! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6508-pull_request-1159

rsteph-de committed:
MCR-2340 add label support for mods classification xslt3 function (#1141)

rsteph-de committed:
MCR-2348 create MCRXSLTransformer with a special TransformerFactory (#1168)

Co-authored-by: Thomas Scheffler

MyCoRe-Travis commented:
Something gone wrong on 6791d6c26ee3abd8d00801310ead01058096c8e2! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6537-pull_request-1159

sebhofmann committed:
MCR-2140 revert "return element always with position bracket ... (#1120)"

This reverts commit e1d8248d

fluetze committed:
MCR-2140 Make XPaths unique in editors

* Fixed buildXPath() to always use pos predicates for elements
* Updated tests
* Rename tests to be more specific
* Checkboxes may be multiple, xpath [1] still means "all"
* Fixed select multiple handling and added test
* Fixed resubmission check for select multiple values
* Fixed unused imports

golsch committed:
change super/guest equals to userid

OPNA2608 committed:
MCR-2338 login NullPointerException if you leave password empty

sebhofmann committed:
MCR-2350 mycore-viewer prevent unnecessary draws of canvas

golsch committed:
MCR-2337 Allow custom user attributes to be stored in JWT token (#1171)

* added querystring based approach to add userattributes to jwt

eckstueck committed:
MCR-1867 On small screens one cannot access all commands

fixed dropdown

eckstueck committed:
MCR-2345 Viewer is sending request for same tiles multiple times (#1162)

* MCR-2345 Viewer is sending request for same tiles multiple times

removed abortLoadingTiles for buffer redraw

* Fixed tile load abort

Co-authored-by: Sebastian Hofmann <sebastian.hofmann@gbv.de>

toKrause committed:
MCR-2331 include email in user search (#1176)

* MCR-2331 Add methods without mailPattern parameter for backwards compatibility
* MCR-2331 Refrain from using Optional method parameters and improve code readability

yagee-de committed:
MCR-2346 provide a test to check priorities of Closeable (#1167)

* MCR-2346 provide a test to check priorities of Closeable
* MCR-2346 keep old behaviour of preparing and closing

yagee-de committed:
MCR-2354 allow do deactivate a PI service

OPNA2608 committed:
MCR-2339 login ArrayIndexOutOfBoundException if you enter a Username with @ as last char (#1174)

Return null (invalid) if username contains @ but splits in more/less than 2 parts

sebhofmann committed:
MCR-2333-commands_to_process_commands_file_from_Classpath (#1137)

MyCoRe-Travis commented:
Something gone wrong on 42f9ab7120009c010e13a3c554a8c8b6039b146a! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6627-pull_request-1159
